_posts/reflect-92.md

```diff
@@ -0,0 +1,202 @@
+                license = @License(name = "MIT Licence", url = "https://github.com/thombergs/code-examples/blob/master/LICENSE")),
+        servers = @Server(url = "http://localhost:8080")
+)
+```
+
+Now, start the server and navigate to: http://localhost:8080/swagger-ui/index.html?configUrl=/reflectoring-openapi/swagger-config
+
+You can see now the information that we defined above!
+![General Information](/assets/img/posts/reflect92/general-info.png)
+
+#### Define Rest API
+
+In this section, we will define our Rest endpoint.
+
+```java
+@RequestMapping("/api/todos")
+@Tag(name = "Todo API", description = "euismod in pellentesque massa placerat duis ultricies lacus sed turpis")
+interface TodoApi {
+
+    @GetMapping
+    @ResponseStatus(code = HttpStatus.OK)
+    List<Todo> findAll();
+
+    @GetMapping("/{id}")
+    @ResponseStatus(code = HttpStatus.OK)
+    Todo findById(@PathVariable String id);
+
+    @PostMapping
+    @ResponseStatus(code = HttpStatus.CREATED)
+    Todo save(@RequestBody Todo todo);
+
+    @PutMapping("/{id}")
+    @ResponseStatus(code = HttpStatus.OK)
+    Todo update(@PathVariable String id, @RequestBody Todo todo);
+
+    @DeleteMapping("/{id}")
+    @ResponseStatus(code = HttpStatus.NO_CONTENT)
+    void delete(@PathVariable String id);
+}
+```
+
+The `@Tag` annotation simple describes the endpoint.
+
+Implementing this interface and annotating our controller with 
+`@RestController` will let springdoc know that this is a controller and should produce a documentation for it.
+```java
+@RestController
+class TodoController implements TodoApi {
+  // omit implementation, not relevant    
+}
+```
+
+Navigate again to http://localhost:8080/swagger-ui/index.html?configUrl=/reflectoring-openapi/swagger-config and take a quick look.
+
+![Todo API Information](/assets/img/posts/reflect92/todo-api-info.png)
+
+Springdoc did its magic and created us the documentation for our API!
+
+Let's dive a little more on Springdoc by defining also a security scheme.
+
+#### Define Security Scheme
+
+To define a Security Scheme for our application we just need to add the `@SecurityScheme` annotation:
+
+```java
+@SecurityScheme(name = "api", scheme = "basic", type = SecuritySchemeType.HTTP, in = SecuritySchemeIn.HEADER)
+```
+
+The above `@SecurityScheme` will be referred as "api" and will do a **basic** authentication via **http**
+
+Let's see what this annotation produced for us:
+
+![Secure Scheme](/assets/img/posts/reflect92/secure-scheme.png)
+
+Our documentation has now also an Authorize Button! If we press this button we will the dialog where we can authenticate.
+
+![Secure Scheme Dialog](/assets/img/posts/reflect92/secure-scheme-dialog.png)
+
+To define that an api endpoint uses the above scheme we have to define it with the `@SecurityRequirement` annotation:
+
+```java
+@SecurityRequirement(name = "api")
```

Comment:
- [ ] Please add the class and method to this example so it would compile. Add a "// other methods omitted" comment for stuff that's not relevant here.